File: Unity/Assets/Eruru.MVVM/Scripts/Shared Project/Bindings/MVVMBindingRelativeSource.cs
using System;

namespace Eruru.MVVM {

	public class MVVMBindingRelativeSource {

		public MVVMBindingRelativeSourceMode Mode { get; set; }
		public Type AncestorType { get; set; }
		public int AncestorLevel { get; set; }

		public MVVMBindingRelativeSource () {
			Mode = MVVMBindingRelativeSourceMode.Self;
		}
		public MVVMBindingRelativeSource (Type ancestorType) {
			AncestorType = ancestorType;
			Mode = MVVMBindingRelativeSourceMode.FindAncestor;
			AncestorLevel = 1;
		}
		public MVVMBindingRelativeSource (Type ancestorType, int ancestorLevel) {
			AncestorType = ancestorType;
			AncestorLevel = ancestorLevel;
			Mode = MVVMBindingRelativeSourceMode.FindAncestor;
		}

	}

}

File: IntelliJ IDEA/Java/src/com/eruru/mvvm/MVVMBindingRelativeSource.java
package com.eruru.mvvm;

import java.io.Serializable;

public class MVVMBindingRelativeSource implements Serializable {

	MVVMBindingRelativeSourceMode mode;
	Class<?> ancestorType;
	int ancestorLevel;

	public MVVMBindingRelativeSource () {
		mode = MVVMBindingRelativeSourceMode.Self;
	}

	public MVVMBindingRelativeSource (Class<?> ancestorType) {
		this (ancestorType, 1);
	}

	public MVVMBindingRelativeSource (Class<?> ancestorType, int ancestorLevel) {
		this.ancestorType = ancestorType;
		this.ancestorLevel = ancestorLevel;
		mode = MVVMBindingRelativeSourceMode.FindAncestor;
	}

}